### ISSUE TITLE
Must ID be an autoincrementing integer? Can it be a cuid/uuid?

### ISSUE THREAD

**Issue by canadaduane** [adamjkb/bark#97]
I notice that the documentation is slightly ambiguous about this point, and I'm curious if there is a way around it. As far as I can tell, the `path` column follows a zero-padded representation of the `id` column based on the current node and its ancestors. However, our tables use [cuid](https://www.npmjs.com/package/@bugsnag/cuid) for the primary key. Does this mean we cannot use bark for these tables?

**Comment by adamjkb** [comment]
Good point! IDs can be of any type as long as they're unique of course ;)
